### PR TITLE
Automate to Automate HA Migration docs

### PR DESCRIPTION
**Pull request by SanjuPal01**
### :nut_and_bolt: Description: What code changed, and why?
Updated the public documentation for Migration of Chef Automate to Automate HA.

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-209

### :athletic_shoe: How to Build and Test the Change
Please check the netlify preview in the PR

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by kalroy — APPROVED** [review]

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 1]
```diff
@@ -18,52 +16,193 @@ gh_repo = "automate"
...
-
-- Chef Automate user running Chef Infra Server in external mode should not migrate to Automate HA.
+- The Minimum version of Automate to perform migration is 4.x.
+- Customers in 2022xxxx or 3.x version needs to update to 4.x version.
+- Automate HA with AWS managed needs to use OpenSearch version <1.3.2
+- The OpenSearch version in Automate should be the same version as Automate HA.
 
 {{< /warning >}}
 
 ## Upgrade with FileSystem Backup Locally
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:26]: update to `Migration`

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 2]
```diff
@@ -18,52 +16,193 @@ gh_repo = "automate"
 
 {{< warning >}}
 
-- Standalone Chef Automate or Chef Automate with embedded Chef Infra Server can migrate to Automate HA, with  
-minimum version of Chef Automate: [20201230192246](https://docs.chef.io/release_notes_automate/#20201230192246)
-
-- Chef Automate user running Chef Infra Server in external mode should not migrate to Automate HA.
+- The Minimum version of Automate to perform migration is 4.x.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:19]: please mark the min version as 2020xxxxxxxx

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 3]
```diff
@@ -18,52 +16,193 @@ gh_repo = "automate"
...
+- Customers in 2022xxxx or 3.x version needs to update to 4.x version.
+- Automate HA with AWS managed needs to use OpenSearch version <1.3.2
+- The OpenSearch version in Automate should be the same version as Automate HA.
 
 {{< /warning >}}
 
 ## Upgrade with FileSystem Backup Locally
 
-1. Create Backup of Chef Automate Standalone using the following command:
+Follow the below steps for Automate HA when you are upgrading with On-Premises and AWS deployment (but not for AWS managed services).
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:28]: This statement is not meaningful

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 4]
```diff
@@ -18,52 +16,193 @@ gh_repo = "automate"
...
-    - Once the backup is completed, save the backup Id. For example: `20210622065515`.
+    Once the backup is completed, save the backup Id. For example: `20210622065515`.
+
+1. If you haven't specified the location in the `config.toml` file, go to the `/var/opt/chef-automate/backups` location and create **Bundle** using the following command:
+
+    ```bash
+    tar -cvf backup.tar.gz <backup_id>/ automatebackup-elasticsearch/ .tmp/
+    ```
+
+1. Transfer the `tar` bundle to one of the Chef Automate HA Frontend Nodes using the following command:
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:52]: We have to transfer the file to Automate node, not the fronted node. Infact we have to do the restoration in Bastion node.

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Review by dishanktiwari2501 — APPROVED** [review]
Reviewed and Updated the Doc

**Review by Dmaddu — APPROVED** [review]

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 5]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
+        Once the backup is completed, save the backup Id. For example: `20210622065515`.
+
+    1. Run the below command to create the `bootstrap.abb` bundle. This bundle captures any local credentials or secrets not persisted in the database.
 
-1. Create Backup of Chef Automate Standalone using the following command:
+        ```bash
+        chef-automate bootstrap bundle create bootstrap.abb
+        ```
+
+1. If you haven't specified the location in the `config.toml` file, go to the `/var/opt/chef-automate/backups` location and create **Bundle** using the following command:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:43]:
```suggestion
1. Go to the backup location mentioned in Automate config, but if you haven't specified the location go to the `/var/opt/chef-automate/backups` location and create **Bundle** using the following command:
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 6]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
 
     ```bash
-    chef-automate backup create
-    chef-automate bootstrap bundle create bootstrap.abb
+    tar -cvf backup.tar.gz <backup_id>/ automatebackup-elasticsearch/ .tmp/
+    ```
+
+1. Transfer the `tar` bundle to one of the Chef-Automate nodes of Automate HA using the following command:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:49]:
```suggestion
1. Transfer the `tar` bundle to one of the Chef Automate nodes of Automate HA using the following command:
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 7]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
+
+    ```bash
+        scp -i /path/to/key /path/to/backup-file user@host:/home/user
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:52]:
```suggestion
        scp -i </path/to/key> </path/to/backup-file> <user>@<host>:/home/<user>
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 8]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
+    ```
+
+1. Transfer the `bootstrap.abb` file to all the Chef Automate HA FrontEnd Nodes (both Chef Automate and Chef Infra Server) using the following command:
+
+    ```bash
+        scp -i /path/to/key /path/to/bootstrap.abb user@host:/home/user
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:58]:
```suggestion
        scp -i </path/to/key> </path/to/bootstrap.abb> <user>@<host>:/home/<user>
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 9]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
-1. Transfer the `tar` bundle to one of the Chef Automate HA Frontend Nodes.
+    The following command will patch the configuration in all the Frontend nodes:
 
-1. Transfer the `bootstrap.abb` file to all the Chef Automate HA FrontEnd Nodes (both Chef Automate and Chef Infra Server).
+    ```bash
+    chef-automate config patch --fe <Path to automate.toml>
+    ```
 
-1. Go the Chef Automate HA, where we copied the `tar` file. Unzip the bundle using:
+1. Go to the Chef-Automate node of Automate HA cluster, where we copied the `tar` file. Unzip the bundle using the following:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:93]:
```suggestion
1. Go to the Chef Automate node of Automate HA cluster, where we copied the `tar` file. Unzip the bundle using the following:
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 10]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
 
     ```bash
     tar -xf backup.tar.gz -C /mnt/automate_backups
     ```
 
-1. Stop all the services at frontend nodes in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
+1. Run the following command at the Chef-Automate node of Automate HA cluster to get the applied `config`:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:99]:
```suggestion
1. Run the following command at the Chef Automate node of Automate HA cluster to get the applied `config`:
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 11]
```diff
@@ -78,62 +216,166 @@ minimum version of Chef Automate: [20201230192246](https://docs.chef.io/release_
     {{% automate/char-warn %}}
     {{< /warning >}}
 
+1. Transfer the `bootstrap.abb` file to all the Chef Automate HA FrontEnd Nodes (both Chef Automate and Chef Infra Server) using the following command:
+
+    ```bash
+        scp -i /path/to/key /path/to/bootstrap.abb user@host:/home/user
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:222]:
```suggestion
        scp -i </path/to/key> </path/to/bootstrap.abb> <user>@<host>:/home/<user>
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 12]
```diff
@@ -18,52 +16,192 @@ gh_repo = "automate"
...
+
+        Once the backup is completed, save the backup Id. For example: `20210622065515`.
+
+    1. Run the below command to create the `bootstrap.abb` bundle. This bundle captures any local credentials or secrets not persisted in the database.
+
+        ```bash
+        chef-automate bootstrap bundle create bootstrap.abb
+        ```
+
+1. Run the following command at the Chef-Automate node of Automate HA cluster to get the applied config:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:201]:
```suggestion
1. Run the following command on the Chef Automate node of Automate HA cluster to get the applied config:
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 13]
```diff
@@ -78,62 +216,166 @@ minimum version of Chef Automate: [20201230192246](https://docs.chef.io/release_
...
 1. Unpack the `bootstrap.abb` file on all the Frontend nodes:
 
     Login to Each Frontend Node and then run after copying the `bootstrap.abb` file.
 
     ```bash
-    chef-automate bootstrap bundle unpack bootstrap.abb
+        chef-automate bootstrap bundle unpack bootstrap.abb
+    ```
+
+1. Stop all the frontend instances except where you saved the `current_config.toml` file at automate nodes in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:233]:
```suggestion
1. Stop all the frontend instances except where you saved the `current_config.toml` file on Chef Automate nodes in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 14]
```diff
@@ -78,62 +216,166 @@ minimum version of Chef Automate: [20201230192246](https://docs.chef.io/release_
...
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+    """
+    ```
+
+1. Create a Backup of Chef Automate Standalone using the following command:
 
-1. Create *Backup* of Chef Automate Standalone using the following command:
+    1. Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the `config.toml` file.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:289]:
```suggestion
    1. Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the Standalone Automate config.
```

File: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md [hunk 15]
```diff
@@ -145,24 +387,72 @@ minimum version of Chef Automate: [20201230192246](https://docs.chef.io/release_
...
-
-1. Unpack the `bootstrap.abb` file on all the Frontend nodes. `ssh` to Each Frontend Node and run the following command:
+1. On every Frontend Node, copy the `bootstrap.abb` file and unpack the file using the following command:
 
     ```bash
     chef-automate bootstrap bundle unpack bootstrap.abb
     ```
 
-1. Run the restore command in one of the Chef Automate node in Chef-Automate HA cluster:
+1. Stop all the frontend instances except where you saved the `current_config.toml` file at automate node in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md:396]:
```suggestion
1. Stop all the frontend instances except where you saved the `current_config.toml` file on Chef Automate node in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
```

**Pull request merged**